.github/workflows/commit_command_checker.yaml

```diff
@@ -0,0 +1,35 @@
+name: 'Check that commit commands have been used correctly'
+on:
+  workflow_call:
+
+jobs:
+  commit-command-checker:
+    runs-on: ubuntu-latest
+
+    steps:
+      - name: Check that 'main' is not merged into the development branch
+        uses: gsactions/commit-message-checker@v2
+        with:
+          excludeDescription: 'true'  # optional: this excludes the description body of a pull request
+          excludeTitle: 'true'  # optional: this excludes the title of a pull request
+          checkAllCommitMessages: 'true'  # optional: this checks all commits associated with a pull request
+          accessToken: ${{ secrets.GITHUB_TOKEN }}  # github access token is only required if checkAllCommitMessages is true
+          pattern: ^(?!Merge branch 'main')
```

Comment:
The pattern is _not_ "Merge branch 'main' into", because this is generic enough to capture any `git pull origin main` command run from the command-line locally before pushing.